[#14] 상품 등록 기능 구현

ethan-sh-lee opened the pull request.

f-lab-dev commented on src/main/java/com/project/smartstore/dto/ProductDto.java:
```diff
@@ -0,0 +1,23 @@
+@AllArgsConstructor
+public class ProductDto {
+
+  String name;
```
접근제한자를 붙여주지 않은 이유가 있으신가요?

ethan-sh-lee replied:
특별한 이유는 없습니다.
하지만 private을 추가하여 불변객체로 만들어 데이터가 개발자의 의도와 다르게 변경되는 것을 방지하는 것이 좋을 것 같습니다.

f-lab-dev commented on src/main/java/com/project/smartstore/service/ProductServiceImpl.java:
```diff
@@ -0,0 +1,18 @@
+
+@Service
+@RequiredArgsConstructor
+public class ProductServiceImpl implements ProductService {
```
`~~Service` -> `~~ServiceImpl` 구조를 가져가는 이유가 무엇이신가요?

ethan-sh-lee replied:
ProductService인터페이스로 추상화하여 구현에 의존하지 않도록 추가하였지만 ProductService인터페이스를 구현한 다른 구현체를 만들 이유가 없기 때문에 굳이 필요하지 않은 것 같습니다.

ethan-sh-lee committed:
기능 : ProductDto 접근제한자 추가

불변 객체로 만들어 데이터가 변하는 위험을 줄이기 위해 접근제한자를 추가하였습니다.

ethan-sh-lee committed:
리팩토링 : ProductService 인터페이스 삭제

ProductService인터페이스를 삭제하고 구현체인 ProductServiceimpl를 ProductService로 변경하여 필요없는 service, serviceimpl를 리팩토링하였습니다.
